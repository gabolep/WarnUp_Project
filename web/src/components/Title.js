import { Component } from 'react';

const styles = {
    title: {
        fontWeight: '630',
        fontSize: '4rem',
    },
    subtitle:{
        fontWeight: '300',
        fontSize: '1.3rem',
        paddingLeft: '5px',
    }
}

class Title extends Component{
    render(){
        return(
            <div>
                <div style = {styles.title}>
                    HN Feed
                </div>
                <div style = {styles.subtitle}>
                    We {'<3'} hacker news!
                </div>
            </div>
            
            
        ) 
    }
}

export default Title