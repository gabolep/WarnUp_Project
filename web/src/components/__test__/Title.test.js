import react from "react";
import ReactDOM from "react-dom";
import Title from '../Title';
import { render, cleanup, screen } from '@testing-library/react'
import renderer from "react-test-renderer";
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup);

describe('<Title />', () =>{
        
    let component
    beforeEach(() => {
        component = render(<Title/>)
    })

    test('renders its children', () => {
        component.getByText('HN Feed')
        component.getByText('We <3 hacker news!')
    })
    test('the style is fine', () => {
        const title = component.getByText('HN Feed')
        const subtitle = component.getByText('We <3 hacker news!')
        expect(title).
        toHaveStyle(
            'font-weight: 630'
        );
        expect(subtitle).toHaveStyle(
            'font-weight: 300'
        )
    })
})
it("matches snapshot", () => {
    const tree = renderer.create(<Title/>).toJSON();
    expect(tree).toMatchSnapshot();
})
