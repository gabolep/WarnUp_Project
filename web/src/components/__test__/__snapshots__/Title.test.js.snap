// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "fontSize": "4rem",
        "fontWeight": "630",
      }
    }
  >
    HN Feed
  </div>
  <div
    style={
      Object {
        "fontSize": "1.3rem",
        "fontWeight": "300",
        "paddingLeft": "5px",
      }
    }
  >
    We 
    &lt;3
     hacker news!
  </div>
</div>
`;
